### PR TITLE
Add a basic demo of integrating data loading and routing

### PR DESCRIPTION
**Pull request by rictic**
Uses just iron-ajax and paper-spinner.

Feedback super welcome!

### CODE REVIEW

File: demo/data-loading-demo/flickr-image-page.html [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+          format: 'json',
+          nojsoncallback: 1
+        };
+      },
+
+      _computeSrc: function(photo) {
+        if (!photo || !photo.farm) {
+          return '';
+        }
+        return 'https://farm' + photo.farm + '.staticflickr.com/' + photo.server + '/' + photo.id + '_' + photo.secret + '.jpg';
```
Comment (e111077) [inline demo/data-loading-demo/flickr-image-page.html]: Line ending style gripe.

**Comment by e111077** [comment]
Other than a style gripe, I don't see anything technical wrong with the demo. Thanks for moving the files into folders!

File: demo/data-loading-demo/flickr-image-page.html (hunk 1 above)
Comment (rictic) [inline demo/data-loading-demo/flickr-image-page.html]: Oops. Fixed this and a couple other places.

**Comment by rictic** [comment]
PTAL!

**Comment by e111077** [comment]
LGTM, but I'd like to see what @robdodson thinks of the demo.

**Comment by robdodson** [comment]
Yeah I think Peter and I were chatting about this on Slack. Seems cool to me

**Comment by BartKoppelmans** [comment]
Any chance this demo could be merged? I am very curious 😁

**Comment by e111077** [comment]
@rictic demo works after migration from carbon -> app

**Comment by rictic** [comment]
Added one more fix so that the redirect from not having query params to having the default search query param of `spaceship` worked. We only want to modify that default search *after* the route is active, so that the query params change is propagated.

**Comment by e111077** [comment]
alles ist gut! :beers: LGTM, do the thing with the green button!

**Pull request merged**